### ISSUE TITLE
Should literal_only_boolean_expressions flag while(true) ?

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#453]
I think the following `while (true)` shouldn't be flagged by `literal_only_boolean_expressions` : 

```dart
/// Return an unused TCP port number.
Future<int> findAvailablePort() async {
  int port = 20000;
  while (true) { // LINT
    try {
      ServerSocket socket =
          await ServerSocket.bind(InternetAddress.LOOPBACK_IP_V4, port);
      await socket.close();
      return port;
    } catch (_) {
      port++;
    }
  }
}
```

**Comment by a14n** [comment]
Perhaps it'd be worth to create another rule to ensure the loop can be exited (by a `break`, `return`, `throw`, `rethrow`...)

**Comment by trinarytree** [comment]
+1 to the original post.
the justification [here](http://dart-lang.github.io/linter/lints/literal_only_boolean_expressions.html) for the current lint rule includes flawed logic.  it says,
"If the condition cannot evaluate to anything but true, the conditional statement is completely redundant."
no, not in the case of `while (true)` and `do while (true)`.
i can't think of any other case, so it seems reasonable to make this into a special case for this lint rule.

perhaps the topic of the 2nd post should go into its own issue.

**Comment by zoechi** [comment]
See also https://github.com/dart-lang/sdk/issues/34181#issuecomment-414125985

> (For the for loop, omitting the condition is the traditional way to make an infinite loop, not writing `true`).

**Comment by pq** [comment]
I lean towards the simple solution, allowing `while(true)` and the related `for` loop.  Would love to hear any reservations though.

/cc @bwilkerson

**Comment by bwilkerson** [comment]
The while and do loops seem like reasonable exceptions, but I've also always seen the condition omitted from a for loop, so I'm hesitant to encourage that case.

**Comment by kevmoo** [comment]
At a minimum, we should add an "Google" example of using `for (;;)` instead of `while (true)`

**Comment by jamesderlin** [comment]
FWIW, preferring `for (;;)` seems backward to me.  When the loop condition is some non-constant expression, I generally use `for` for *definite* loops and `while` for *indefinite* ones.  Situations where I'd want an "infinite" loop are invariably indefinite cases.
